web/client-api/src/main/java/io/deephaven/web/client/api/widget/JsWidget.java

```diff
@@ -0,0 +1,46 @@
+package io.deephaven.web.client.api.widget;
+
+import elemental2.core.Uint8Array;
+import io.deephaven.javascript.proto.dhinternal.io.deephaven.proto.object_pb.FetchObjectResponse;
+import io.deephaven.web.client.api.WorkerConnection;
+import jsinterop.annotations.JsMethod;
+import jsinterop.annotations.JsProperty;
+
+public class JsWidget {
+    private final WorkerConnection connection;
+
+    private final FetchObjectResponse response;
+
+    private JsWidgetExportedObject[] exportedObjects;
+
+    public JsWidget(WorkerConnection connection, FetchObjectResponse response) {
+        this.connection = connection;
+        this.response = response;
+    }
+
+    @JsProperty
+    public String getType() {
+        return response.getType();
+    }
+
+    @JsMethod
+    public String getDataAsBase64() {
```

Comment:
for json at least, i'd still vote a getDataAsUtf8 or something like that? just to do it internally nicely, rather than have to read the bytes and decode externally

Reply:
TextDecoder isn't in current version of GWT, I think need to wait for #1860 to merge first?

Reply:
Yeah let's wait, note added.